packages/data/src/components/registry-provider/index.js

```diff
@@ -0,0 +1,10 @@
+/**
+ * WordPress dependencies
+ */
+import { createContext } from '@wordpress/element';
+
+const { Consumer, Provider } = createContext( null );
```

Comment:
Rather than having `withDispatch` and `withSelect` know about the default registry, could we just assign it as the default here instead of `null`? Avoids need for special handling, where:

```js
const dispatch = this.props.registry ? this.props.registry.dispatch : defaultRegistry.dispatch;
```

Becomes:

```js
const { dispatch } = this.props.registry;
```

Reply:
I think the issue is that people use `withSelect`/ `withDispatch` without using a provider. I'm wondering if the default value is stored in the provider or can be provided by the consumer even without any provider. I'll have to check.

Reply:
I _think_ the consumer should inherit the default value if there is no ascendant provider.

Reply:
Demo: https://codepen.io/aduth/pen/rKZrKo

Reply:
> I think the consumer should inherit the default value if there is no ascendant provider.

We use this behavior both in tests and for `Disabled` component (see #7138). This is why you have to provide this default value when creating context.

Reply:
If I follow properly, the following should do the trick:

```js
const { Consumer, Provider } = createContext( createRegistry() );
```